src/modules/gateway/sagas.ts

```diff
@@ -83,268 +100,341 @@ export function createGatewaySaga(config: ManaFiatGatewaySagasConfig) {
     }
   }
-}
-
-function* handleOpenBuyManaWithFiatModal(
-  config: ManaFiatGatewaySagasConfig,
-  action: OpenBuyManaWithFiatModalRequestAction
-) {
-  try {
-    const { selectedNetwork } = action.payload
-    const pendingManaPurchase: Purchase | undefined = yield select(
-      getPendingManaPurchase
-    )
 
-    if (pendingManaPurchase) {
-      let goToUrl: string | undefined
+  function* handleOpenFiatGatewayWidget(
+    action: OpenFiatGatewayWidgetRequestAction
+  ) {
+    const { data, gateway, listeners } = action.payload
+    try {
+      switch (gateway) {
+        case FiatGateway.WERT:
+          const { onLoaded, onPending, onSuccess } = listeners || {}
+          const { marketplaceServerURL } = config[FiatGateway.WERT]
+
+          const wallet: Wallet | null = yield select(getWalletData)
+          if (wallet) {
+            const identity: AuthIdentity | null = yield call(
+              getIdentityOrRedirect
+            )
 
-      if (pendingManaPurchase.gateway === NetworkGatewayType.MOON_PAY) {
-        goToUrl = new MoonPay(config.moonPay).getTransactionReceiptUrl(
-          pendingManaPurchase.id
-        )
+            if (!identity) {
+              yield put(openFiatGatewayWidgetFailure(NO_IDENTITY_ERROR))
+              return
+            } else if (identity) {
+              const isDev = getEnv() === Env.DEVELOPMENT
+              const {
+                commodity,
+                commodity_amount,
+                sc_address,
+                sc_input_data
+              } = data
+              if (
+                commodity &&
+                commodity_amount &&
+                sc_address &&
+                sc_input_data
+              ) {
+                const dataToSign: WertMessage = {
+                  address: wallet.address,
+                  commodity,
+                  commodity_amount,
+                  network: isDev ? 'sepolia' : 'ethereum', // will be wallet.network
+                  sc_address,
+                  sc_input_data
+                }
+
+                const marketplaceAPI = new MarketplaceAPI(marketplaceServerURL)
+
+                const signature: string = yield call(
+                  [marketplaceAPI, 'signWertMessage'],
+                  dataToSign,
+                  identity
+                )
+
+                const wertWidget = new WertWidget({
+                  ...data,
+                  ...dataToSign,
+                  signature,
+                  listeners: {
+                    loaded: onLoaded,
+                    'payment-status': options => {
+                      if (options.tx_id) {
+                        // it's a success event
+                        onSuccess?.({ data: options, type: 'payment-status' })
+                      } else {
+                        onPending?.({ data: options, type: 'payment-status' })
+                      }
+                    }
+                  }
+                })
+
+                wertWidget.open()
```

Comment:
Would you mind wrapping the this line with a `yield call`? If not, the code will try to instantiate the modal and that might be time consuming.
```suggestion
                yield call([wertWidget, ['open'])
```